src/Nethermind/Nethermind.State/StorageTree.cs

```diff
@@ -45,11 +46,11 @@ public StorageTree(IScopedTrieStore? trieStore, Hash256 rootHash, ILogManager? l
             TrieType = TrieType.Storage;
         }
 
-        private static void GetKey(in UInt256 index, in Span<byte> key)
+        private static void GetKey(in UInt256 index, ref Span<byte> key)
```

Comment:
We could avoid allocating the key span in the calling method when we hit the cache, can't we?
I would feel safer that if we hit the cache we would return `ReadOnlySpan`.
Caller methods need to be reworked to achieve both.